test/eoftests/eoftests_runner.cpp

```diff
@@ -0,0 +1,107 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2022 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "eoftests.hpp"
+#include <evmc/evmc.hpp>
+#include <evmone/eof.hpp>
+#include <gtest/gtest.h>
+
+
+namespace evmone::test
+{
+namespace json = nlohmann;
+void run_eof_test(std::istream& input)
+{
+    const EOFTests test = json::json::parse(input).get<EOFTests>();
+    for (const auto& cases : test.cases)
+    {
+        for (const auto& expectation : cases.expectations)
+        {
+            const EOFValidationError result = evmone::validate_eof(expectation.rev, cases.code);
+            bool b_result = true;
+            if (result != EOFValidationError::success)
+            {
+                b_result = false;
+            }
+            // else
+            //{
+            //     EXPECT_EQ(get_error_message(result), expectation.exception);
+            // }
+            EXPECT_EQ(b_result, expectation.result);
+        }
+    }
+}
+
+evmc_revision to_rev(std::string_view s)
```

Comment:
We shouldn't duplicate this function. Please move it to `test/utils` and make it shared between this tool and statetest